### ISSUE TITLE
Refactor Openshift extension on top of Kubernetes

### ISSUE THREAD

**Issue by jwendell** [arquillian/arquillian-cube#483]
Making it extends kubernetes avoids code duplication and reflects what in fact Openshift is: an enhanced Kubernetes.

**Comment by lordofthejars** [comment]
Yes this is something that @iocanel wants to work as well i two steps first one:

https://github.com/arquillian/arquillian-cube/issues/445

And second one by aligning kubernetes objects to Cube interfaces too.

@iocanel @jwendell go ahead, if you need anything please feel free to ping me or @bartoszmajsak

**Comment by iocanel** [comment]
If there are no objections I am on it.

**Comment by jwendell** [comment]
@iocanel no problem. I had started to play with it, didn't get too far, but surprisingly that was not a huge refactoring as I was thinking. I managed to run a simple "hello world" test with a small set of changes.

Here is what I have done so far:
https://github.com/jwendell/arquillian-cube/commit/38d2bcc2c95907e92089ee25af7c1dfcfb920278

Feel free to use it, or some part of it, or discard it at all.

The next step I was about to do was to move the port forwarding code from openshit to kubernetes, as this is a kubernetes feature. If you want, I can do this separately and provide a PR with just this change.

**Comment by iocanel** [comment]
@jwendell since you started, then please ho for it :-)

**Comment by lordofthejars** [comment]
@jwendell happy to hear about that, when you finish this movement, we can start planning to do https://github.com/arquillian/arquillian-cube/issues/468 https://github.com/arquillian/arquillian-cube/issues/467 and https://github.com/arquillian/arquillian-cube/issues/465

**Comment by jwendell** [comment]
for reference, my working can be tracked here: https://github.com/jwendell/arquillian-cube/commits/483
comments are welcome!

**Comment by lordofthejars** [comment]
@jwendell @iocanel I have reviewed that there are several issues interrelated between them about integration of kubernetes/openshift (or even a PR https://github.com/arquillian/arquillian-cube/pull/493) how we can proceed about this?

**Comment by iocanel** [comment]
My main issue is that currently there are tests that are broken in the openshift extension so I can't really do any kind of refactoring (I can't know if I broke something or not).

So we can either fix tests and refactor later, or ignore tests, refactor and fix later if needed.

**Comment by iocanel** [comment]
@lordofthejars: fwiw i created #561 which for now seems to build / test with no issues.

**Comment by lordofthejars** [comment]
I can close this issue since PR is merged isn't it?
